mesonbuild/interpreter/interpreter.py

```diff
@@ -1958,6 +1958,12 @@ def func_custom_target(self, node: mparser.FunctionNode, args: T.Tuple[str],
             self.validate_forbidden_targets(t)
         self._validate_custom_target_outputs(len(inputs) > 1, kwargs['output'], "custom_target")
 
+        output_set = set(kwargs['output'])
+        if len(output_set) != len(kwargs['output']):
+            for el in output_set:
+                if kwargs['output'].count(el) > 1:
```

Comment:
This is ruinously inefficient to repeatedly walk the entire list. I would instead suggest walking the list *once*, and alternately appending to either `seen` or `duplicates`